dev/com.ibm.ws.security.kerberos.auth/src/com/ibm/ws/security/kerberos/auth/internal/LRUCache.java

```diff
@@ -84,37 +86,38 @@ public Subject get(KerberosPrincipal principal) {
 
             // Ensure the KerberosTicket is current and attempt to renew if not
             Set<KerberosTicket> kerbTickets = sub.getPrivateCredentials(KerberosTicket.class);
-            KerberosTicket renewedTicket = null;
-            KerberosTicket oldTicket = null;
             for (KerberosTicket kerbTicket : kerbTickets) {
-                if (!kerbTicket.isCurrent()) {
-                    if (kerbTicket.isRenewable()) {
-                        // If the ticket is not current but is renewable, try to renew the ticket if
-                        // there is at least 10m remaining on the renewTill time
-                        long renewTillMs = kerbTicket.getRenewTill().getTime();
-                        long timeLeft = renewTillMs - System.currentTimeMillis() - TimeUnit.MINUTES.toMillis(10);
-                        if (timeLeft < 0) {
+                if (kerbTicket.isCurrent()) {
+                    // we have a valid ticket
+                    break;
```

Comment:
If the kerberos ticket is current but almost expired, should we try to refresh it?

Reply:
yes good point, I'll fix this